### PR TITLE
feat: Add connect telegram and x functionality with hi-fi design

### PR DESCRIPTION
**Pull request by yash431garg**
- Description
User will be able to link/unlink x and telegram account from their connected wallet address simultaneously updating the database. 
- Screen shots or recordings for UI changes
Screenshot 2024-11-21 at 5 07 49 PM

### CODE REVIEW

File: app/actions/user.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"use server";
```
Comment (marvinmarnold) [inline app/actions/user.ts:1]: Can you rename the file to addUserSocials?

File: app/components/SocialAuth/SocialAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          onClick={async () => {
+            await linkSocialAccount(ProviderEnum.Twitter);
+          }}
+          className="bg-gray-700 hover:bg-gray-700"
+        >
+          <XIconWhite width={14} height={14} />
+          <p className="pl-4">Connect X</p>
+        </Button>
+      )}
+      {userTelegramInfo ? (
```
Comment (marvinmarnold) [inline app/components/SocialAuth/SocialAuth.tsx:83]: Doesn't look like this follows the AC "It should not be possible to disconnect TG if user signed up through the bot and there is no other wallet associated with the account."

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
Screenshot 2024-11-21 at 4 26 51 PM

I'm not able to load the login screen for some reason.

File: app/components/SocialAuth/SocialAuth.tsx (hunk 2 above)
Comment (yash431garg) [reply app/components/SocialAuth/SocialAuth.tsx:83]: I think I missed that.

**Comment by yash431garg** [comment]
> * Description
>   User will be able to link/unlink x and telegram account from their connected wallet address simultaneously updating the database.
> * Screen shots or recordings for UI changes
> 
> Screenshot 2024-11-21 at 5 07 49 PM

Done it was because of missing db migration. Now it should be working.

**Comment by yash431garg** [comment]
> Done it was because of missing db migration. Now it should be working.

Done it was because of missing db migration. Now it should be working.

**Comment by yash431garg** [comment]
> Screenshot 2024-11-21 at 4 26 51 PM
> 
> I'm not able to load the login screen for some reason.

Done it was because of missing db migration. Now it should be working.

**Comment by marvinmarnold** [comment]
> Resolved it was because of missing db migration. Now it should be working.

Oh ok. This points out that we don't have a very clear process for how to handle migrations in staging. Some thoughts/questions

1. To fix the issue, did you manually run the migrations on staging? In some cases, that will cause the GH workflow to fail because it will try to reapply a duplicate change like "create index". It also gives us less confidence that the subsequent release to prod will work as expected.
2. An alternative approach is to first review the code, then merge into staging and test again there after migrations run. In this case, devs should just note that preview link wont work. 
3. We could also consider implementing some better dev infra. For example DB branching so there's a separate DB per preview link.

I think #2 is fine for now. It's pretty risk free to test things in staging.

 cc @samott

**Comment by yash431garg** [comment]
> > Resolved it was because of missing db migration. Now it should be working.
> 
> Oh ok. This points out that we don't have a very clear process for how to handle migrations in staging. Some thoughts/questions
> 
> 1. To fix the issue, did you manually run the migrations on staging? In some cases, that will cause the GH workflow to fail because it will try to reapply a duplicate change like "create index". It also gives us less confidence that the subsequent release to prod will work as expected.
> 2. An alternative approach is to first review the code, then merge into staging and test again there after migrations run. In this case, devs should just note that preview link wont work.
> 3. We could also consider implementing some better dev infra. For example DB branching so there's a separate DB per preview link.
> 
> I think #2 is fine for now. It's pretty risk free to test things in staging.
> 
> cc @samott

Yes, I manually run the migrations. If we go with #2 Will we be testing the features in staging link and not in the preview branch.

**Comment by marvinmarnold** [comment]
Old comment:

@yash431garg I'm still getting these errors

- Twitter error
Screenshot 2024-11-22 at 9 24 10 PM
- TG not working
Screenshot 2024-11-22 at 9 25 15 PM

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
@yash431garg this looks like it's working. but some follow ups:
- Can you add details in README or Notion about how to get X and Telegram keys necessary for setup? also about how to configure Dynamic correctly.
- Depending on the steps above, we may need to put one or both account linking options behind and env var feature flag.
- I'll have to test that tg can't be unlinked when sign up happens through bot after you merge into staging. 

> Yes, I manually run the migrations. If we go with https://github.com/gator-labs/chomp/pull/2 Will we be testing the features in staging link and not in the preview branch.

Yup. That's not ideal, but I think it's a good enough process for now.

**Comment by yash431garg** [comment]
> @yash431garg this looks like it's working. but some follow ups:
> 
> * Can you add details in README or Notion about how to get X and Telegram keys necessary for setup? also about how to configure Dynamic correctly.
> * Depending on the steps above, we may need to put one or both account linking options behind and env var feature flag.
> * I'll have to test that tg can't be unlinked when sign up happens through bot after you merge into staging.
> 
> > Yes, I manually run the migrations. If we go with #2 Will we be testing the features in staging link and not in the preview branch.
> 
> Yup. That's not ideal, but I think it's a good enough process for now.

I have tested unlink telegram but dynamic throws an error and I have also added a custom error but let me know if you get any error.

**Review by marvinmarnold — APPROVED** [review]
I tested it and it works well. I also created a follow up task for us to discuss: https://linear.app/gator/issue/PROD-504/improve-infra-for-demo-environment

**Comment by marvinmarnold** [comment]
Can you also test that tx signing works if you:

1. sign up with Phantom
2. connect Telegram
3. sign out
4. sign in with Telegram
5. try to reveal

It should still let you use Phantom burn. @neel-ds is this the edge case you were talking about, or did I misunderstand?

**Comment by yash431garg** [comment]
> Can you also test that tx signing works if you:
> 
> 1. sign up with Phantom
> 2. connect Telegram
> 3. sign out
> 4. sign in with Telegram
> 5. try to reveal
> 
> It should still let you use Phantom burn. @neel-ds is this the edge case you were talking about, or did I misunderstand?

I have tested that you can check the comments on linear.

**Comment by yash431garg** [comment]
> > Can you also test that tx signing works if you:
> > 
> > 1. sign up with Phantom
> > 2. connect Telegram
> > 3. sign out
> > 4. sign in with Telegram
> > 5. try to reveal
> > 
> > It should still let you use Phantom burn. @neel-ds is this the edge case you were talking about, or did I misunderstand?
> 
> I have tested that and you can check the comments on linear. https://linear.app/gator/issue/PROD-457/link-telegram-andor-x-account-from-profile#comment-5043b1db

**Review by Yihwan — APPROVED** [review]
